internal/oci/pull.go

```diff
@@ -141,11 +142,19 @@ type RemoteClient struct{}
 // Image fetches an image manifest. The returned image lazily pulls its layers.
 func (i *RemoteClient) Image(ctx context.Context, ref name.Reference, o ...ImageClientOption) (ociv1.Image, error) {
 	opts := parse(o...)
+	if opts.pull == ImagePullPolicyNever {
+		return nil, errors.New(errPullNever)
+	}
 	if opts.auth != nil {
 		return remote.Image(ref, remote.WithContext(ctx), remote.WithAuth(opts.auth))
 	}
-	if opts.pull == ImagePullPolicyNever {
-		return nil, errors.New(errPullNever)
+	keychain, err := k8schain.NewInCluster(ctx, k8schain.Options{})
+	var image ociv1.Image
+	if err == nil {
+		image, err = remote.Image(ref, remote.WithContext(ctx), remote.WithAuthFromKeychain(keychain))
+	}
+	if err == nil {
+		return image, nil
 	}
 	return remote.Image(ref, remote.WithContext(ctx))
```

Comment:
```suggestion
	keychain, err := k8schain.NewInCluster(ctx, k8schain.Options{})
	if err != nil {
		// In-cluster auth not supported. Fall back to no auth.
		return remote.Image(ref, remote.WithContext(ctx))
	}
	img, err := remote.Image(ref, remote.WithContext(ctx), remote.WithAuthFromKeychain(keychain))
	if err != nil {
		// In-cluster auth not supported. Fall back to no auth.	
		return remote.Image(ref, remote.WithContext(ctx))
	}
	return img, nil
```

Would this logic be equivalent? If so I'd prefer it, as I find it easier to follow.